examples/states_ui/main.rs

```diff
@@ -0,0 +1,99 @@
+use amethyst::{
+    assets::{HotReloadBundle, Processor},
+    audio::Source,
+    core::transform::TransformBundle,
+    input::{InputBundle, StringBindings},
+    prelude::*,
+    renderer::{
+        plugins::RenderToWindow,
+        // rendy::mesh::{Normal, Position, TexCoord},
+        types::DefaultBackend,
+        RenderingBundle,
+    },
+    ui::{RenderUi, UiBundle},
+    utils::application_root_dir,
+    utils::fps_counter::FpsCounterBundle,
+};
+
+mod credits;
+mod events;
+mod game;
+mod menu;
+mod pause;
+mod util;
+mod welcome;
+
+/// Quick overview what you can do when running this example:
+///
+/// Switch from the 'Welcome' Screen to the 'Menu' Screen.
+/// From the 'Menu', switch to either 'Credits' (from which you can only switch back) or to 'Game'.
+/// From 'Game', you can enter the 'Pause' menu.
+/// Here you can select to resume (go back to 'Game'), exit to menu (go to 'Menu') or exit (quit).
+///
+/// During the 'Pause' menu, the 'Game' is paused accordingly.
+
+pub fn main() -> amethyst::Result<()> {
+    amethyst::start_logger(Default::default());
+
+    // this will be the directory the 'Cargo.toml' is defined in.
+    let app_root = application_root_dir()?;
+
+    // our display config is in our configs folder.
+    let display_config_path = app_root.join("examples/states_ui/config/display.ron");
+
+    // other assets ('*.ron' files, '*.png' textures, '*.ogg' audio files, ui prefab files, ...) are here
+    let assets_dir = app_root.join("examples/assets");
+
+    let game_data = GameDataBuilder::default()
+        // a lot of other bundles/systems depend on this (without it being explicitly clear), so it
+        // makes sense to add it early on
+        .with_bundle(TransformBundle::new())?
+        // this bundle allows us to 'find' the Buttons and other UI elements later on
+        .with_bundle(UiBundle::<StringBindings>::new())?
+        // this allows us to reload '*.ron' files during execution
+        .with_bundle(HotReloadBundle::default())?
+        // without this System, our Program will silently (!) fail when trying to load the 'Game'.
+        // (try it!)
+        // This System takes care of Audio (in this case, the Button audio for hovering/clicking/...
+        .with(Processor::<Source>::new(), "source_processor", &[])
```

Comment:
Hm, there's an `amethyst::audio::AudioBundle` which adds this, as well as the system to play the sound.

Reply:
Okay, so do you recommend using them instead? How would that be different though?

Reply:
Yeap I'd recommend using the bundle -- it's the "thing that knows what loads the sound, and what plays the sound". It's not completely necessary if you simply wish to allow sounds to be loaded but never played (e.g. the button that plays a sound on hover / mouse click)

Reply:
okay, did that just now